src/TransportCompatibilityTests/AzureServiceBusV8/EndpointFacade.cs

```diff
@@ -0,0 +1,117 @@
+﻿namespace AzureServiceBusV8
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using TransportCompatibilityTests.Common;
+    using TransportCompatibilityTests.Common.AzureServiceBus;
+    using TransportCompatibilityTests.Common.Messages;
+
+    public class EndpointFacade : MarshalByRefObject, IEndpointFacade
+    {
+        IEndpointInstance endpointInstance;
+        MessageStore messageStore;
+        CallbackResultStore callbackResultStore;
+        SubscriptionStore subscriptionStore;
+
+        async Task InitializeEndpoint(AzureServiceBusEndpointDefinition endpointDefinition)
+        {
+            var endpointConfiguration = new EndpointConfiguration(endpointDefinition.Name);
+
+            endpointConfiguration.Conventions().DefiningMessagesAs(t => t.Namespace != null && t.Namespace.EndsWith(".Messages") && t != typeof(TestEvent));
+            endpointConfiguration.Conventions().DefiningEventsAs(t => t == typeof(TestEvent));
+
+            endpointConfiguration.EnableInstallers();
+            endpointConfiguration.UsePersistence<InMemoryPersistence>();
+            var transportConfiguration = endpointConfiguration.UseTransport<AzureServiceBusTransport>();
+            transportConfiguration
+                .UseEndpointOrientedTopology()
```

Comment:
@timbussmann is there a specific reason why u use endpoint oriented topology here? The prefered and new topology is the forwarding one. Endpoint Oriented topology requires publisher registration.

Of course it would be ideal if the topology could be another permutation of the tests.

Reply:
@danielmarbach you're correct. Though ASB v6 supported EndpointOriented topology only.

Reply:
@danielmarbach the tests send messages from/to different versions of the transport. I was assuming that they all need to run on the same topology?

Reply:
@timbussmann that is correct, they have to be